src/codecs/hdr/decoder.rs

```diff
@@ -143,14 +144,25 @@ impl<R: BufRead> HdrAdapter<R> {
     }
 
     /// Read the actual data of the image, and store it in Self::data.
-    fn read_image_data(&mut self, buf: &mut [u8]) -> ImageResult<()> {
-        assert_eq!(u64::try_from(buf.len()), Ok(self.total_bytes()));
+    fn read_image_data_f32(&mut self, image_bytes: &mut [u8]) -> ImageResult<()> {
+        panic!("dsfowsidjw");
```

Comment:
running the tests currently does not trigger this panic. some ideas why this is not triggered?

Reply:
the benchmarks trigger this panic